unblob/file_utils.py

```diff
@@ -386,60 +423,66 @@ def record_problem(self, problem: ExtractionProblem):
     def has_root_permissions(self):
         return os.geteuid() == 0
 
-    def is_safe_path(self, path: Path) -> bool:
-        return path.is_absolute() and is_safe_path(self.root, path)
-
-    def format_path(self, path: Path) -> str:
-        """Convert a potentially absolute path to root relative path.
-
-        Converting to relative paths highlights how path traversal happens.
-        """
-        return os.path.relpath(path, start=self.root)
+    def _fs_path(self, path: Path) -> _FSPath:
+        return _FSPath(root=self.root, path=path)
 
-    def check_path(self, path: Path, operation: str) -> bool:
-        if self.is_safe_path(path):
-            return True
+    def get_checked_path(self, path: Path, path_use_description: str) -> Optional[Path]:
+        fs_path = self._fs_path(path)
+        if fs_path.is_safe:
+            return fs_path.absolute_path
 
         report = ExtractionProblem(
-            path=self.format_path(path),
-            problem=f"Potential path traversal through {operation}",
+            path=fs_path.format_path(),
+            problem=f"Potential path traversal through {path_use_description}",
             resolution="Skipped.",
         )
         self.record_problem(report)
-        return False
+        return None
 
     def write_bytes(self, path: Path, content: bytes):
         logger.debug("creating file", file_path=path, _verbosity=3)
-        if self.check_path(path, "write_bytes"):
-            path.write_bytes(content)
+        safe_path = self.get_checked_path(path, "write_bytes")
+
+        if safe_path:
+            safe_path.write_bytes(content)
```

Comment:
Yes, and it was like that, but changed for the less terse version, because I felt too many things was crammed into the `if` condition, while it should be simple.

Walrus is controversial, and in this case I do not think it is worth it:  more harm is done to readabillity, than it is gained through conciseness.